### PR TITLE
Configure host for global registration

### PR DESCRIPTION
**Pull request by Lennonka**
#### What changes are you introducing?

A host needs to be configured a certain way before it can be registered by using Global Registration and that isn't properly documented at the present time.

We have a _Configure a host for registration_ module, but we don't use it in the _Managing Hosts_ guide currently, which we should. Originally, it covered only the NTP synchronization.

Also, we should provide a prerequisite so that the user can make a secured curl call right away. The CA certificate config is now also included in the pre-registration configuration.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

Because it's better to tell the user how the host must be configured for registration beforehand instead of explaining it when it's "too late".

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

Depends on #3193

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [ ] Foreman 3.11/Katello 4.13
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

**Comment by Lennonka** [comment]
Pushed fix - missing `endif`.

File: guides/common/assembly_registering-hosts.adoc [hunk 1]
```diff
@@ -11,6 +11,10 @@ include::modules/con_registering-hosts-by-using-global-registration.adoc[levelof
 
 include::modules/ref_global-parameters-for-registration.adoc[leveloffset=+2]
 
+ifdef::katello,orcharhino,satellite[]
+include::modules/proc_configuring-a-host-for-registration.adoc[leveloffset=+2]
+endif::[]
```
Comment (Lennonka) [inline guides/common/assembly_registering-hosts.adoc:16]: I'm including this procedure for non-Katello products only because time synchronization wasn't required and I was told that vanilla Foreman doesn't care about the certificate.

**Comment by Lennonka** [comment]
Open to suggestions :)

File: guides/common/modules/proc_registering-a-host.adoc [hunk 2]
```diff
@@ -58,40 +58,14 @@ A {SmartProxy} behind a load balancer takes precedence over a {SmartProxy} selec
...
-+
-To do this, find where {Project} stores the CA file by navigating to *Administer* > *Settings* > *Authentication* and locating the value of the *SSL CA file* setting.
-+
-Copy the CA file to the `/etc/pki/ca-trust/source/anchors/` directory on hosts and enter the following commands:
-+
-[options="nowrap", subs="+quotes,attributes"]
-----
-# update-ca-trust enable
-# update-ca-trust
-----
```
Comment (Lennonka) [inline guides/common/modules/proc_registering-a-host.adoc:80]: My main point was to move this into an action that has to be done as a prerequisite to registration so that the user can make a secure call right away.

File: guides/common/assembly_registering-hosts.adoc (hunk 1 above)
Comment (stejskalleos) [reply guides/common/assembly_registering-hosts.adoc:16]: > for non-Katello products

Isn't it the opposite, `ifdef::katello,orcharhino,satellite[]` is for katello, right?

Comment (Lennonka) [reply guides/common/assembly_registering-hosts.adoc:16]: @stejskalleos That's correct.

File: guides/common/modules/proc_registering-a-host.adoc [hunk 3]
```diff
@@ -4,10 +4,10 @@
 You can register a host by using registration templates and set up various integration features and host tools during the registration process.
 
 .Prerequisites
-* Your user account has a role assigned that has the `create_hosts` permission.
+* Your user account has a role assigned that grants the `create_hosts` permission.
 * You must have root privileges on the host that you want to register.
 ifdef::satellite,orcharhino,katello[]
-* {ProjectServer}, any {SmartProxyServers}, and all hosts must be synchronized with the same NTP server, and have a time synchronization tool enabled and running.
+* You have configured the host for registration.
```
Comment (stejskalleos) [inline guides/common/modules/proc_registering-a-host.adoc:10]: Can we make it as a link to the `guides/common/modules/proc_configuring-a-host-for-registration.adoc` section

File: guides/common/modules/proc_registering-a-host.adoc [hunk 4]
```diff
@@ -58,40 +58,14 @@ A {SmartProxy} behind a load balancer takes precedence over a {SmartProxy} selec
 ifdef::katello,satellite,orcharhino[]
 . In the *Activation Keys* field, enter one or more activation keys to assign to hosts.
 endif::[]
-. Optional: Select the *Insecure* option, if you want to make the first call insecure.
-During this first call, hosts download the CA file from {Project}.
+. Optional: Select the *Insecure* option, if you want to make the registration call insecure.
```
Comment (stejskalleos) [inline guides/common/modules/proc_registering-a-host.adoc:61]: I would rephrase it:
It's not like users want to make it insecure. It's because the CA file is not in the host's "trusted store."
So I would say something like, "If the CA is not trusted by the host, use insecure for the initial call. Then, the registration will download the CA automatically and use a secure connection for the next calls and communication."

Comment (Lennonka) [reply guides/common/modules/proc_registering-a-host.adoc:61]: @stejskalleos That's exactly the situation I'm trying to prevent by having the user to configure the host before registration. The pre-registration config involves adding the CA cert to the trust store. Are there any other use cases for an insecure call?

Comment (maximiliankolb) [reply guides/common/modules/proc_registering-a-host.adoc:61]: If users SSH into a host to run the curl/wget command, then they'd also have a way to securely deploy the Foreman CA cert. With guides/common/modules/proc_configuring-a-host-for-registration.adoc, which I like, there should not be a valid use case for `Insecure` anymore.

I also think the explanation by Leos makes sense; it's IMHO better than "click insecure to make X insecure" -> "Select insecure if your host does not trust the Foreman CA prior to the registration".

File: guides/common/modules/proc_registering-a-host.adoc (hunk 2 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_registering-a-host.adoc:80]: As long as Foreman Web UI offers the Insecure checkbox, we should keep the procedure on how to import the Foreman CA cert. Instead of deleting it, maybe move it to a separate procedure which we could specify as prerequisite?

**Review by maximiliankolb — COMMENTED** [review]
some suggestions;

File: guides/common/modules/proc_configuring-a-host-for-registration.adoc [hunk 5]
```diff
@@ -2,24 +2,42 @@
...
 # service ntpd start
 ----
 endif::[]
+. If you use a self-signed {Project} CA certificate, deploy the CA certificate on your host so that the host can make a secured registration call.
+ifdef::managing-hosts[]
+For more information, see xref:deploying-the-ca-certificate-on-a-host-manually[].
+endif::[]
+ifndef::managing-hosts[]
+{ManagingHostsDocURL}deploying-the-ca-certificate-on-a-host-manually[Deploying the CA certificate ona  host manually] in _{ManagingHostsDocTitle}_.
+endif::[]
```
Comment (Lennonka) [inline guides/common/modules/proc_configuring-a-host-for-registration.adoc:43]: @maximiliankolb FYI, this is the step to configure the CA cert. Since there's a procedure how to do that I'm just linking to it instead of duplicating the content.

Comment (Lennonka) [reply guides/common/modules/proc_configuring-a-host-for-registration.adoc:43]: On second thought, I think I'm mixing up two different certs. The original procedure talks about locating the **SSL CA file** setting, which refers to a different cert.

**Comment by Lennonka** [comment]
I have rewritten the configuring procedure because earlier I was referring users to an incorrect certificate file.

I have also successfully tested the configuration + secure registration on RHEL 9. :heavy_check_mark:

**Review by maximiliankolb — APPROVED** [review]
LGTM style-wise.

**Comment by Lennonka** [comment]
I'm setting the `tech review done` flag because I've tested it successfully. ^^

**Comment by Lennonka** [comment]
Rebased.

**Comment by Lennonka** [comment]
Cherry-picked:
- cd06a1e2b0..de15ce94e8  3.12 -> 3.12
- bc2ad19748..34db4e3ad0  3.11 -> 3.11
- 3d4cead6da..398118deca  3.10 -> 3.10
- caca9c03b6..530deeb3fd  3.9 -> 3.9

There were conflicts on 3.8, therefore I'm not cherrypicking it further.
